### PR TITLE
Saving cluster yaml token to clusters collection

### PR DESCRIPTION
**Pull request by rajdas98**
Signed-off-by: Raj Babu Das <raj.das@mayadata.io>

## Proposed changes

Saving cluster yaml token to clusters collection
## Types of changes

What types of changes does your code introduce to Litmus? Put an `x` in the boxes that apply
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices applies)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
- [ ] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus/blob/master/CONTRIBUTING.md) doc
- [ ] I have signed the commit for DCO to be passed.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added necessary documentation (if appropriate)

## Dependency
- Please add the links to the dependent PR need to be merged before this (if any).

### CODE REVIEW

**Review by oumkale — APPROVED** [review]

**Comment by gdsoumya** [comment]
Why do we need to save the token?

File: litmus-portal/graphql-server/pkg/graphql/mutations/mutation.go [hunk 1]
```diff
@@ -33,15 +41,10 @@ func ClusterRegister(input model.ClusterInput) (*model.ClusterRegResponse, error
...
 		UpdatedAt:    strconv.FormatInt(time.Now().Unix(), 10),
+		Token:        token,
 	}
 
-	err := database.InsertCluster(newCluster)
-	if err != nil {
-		return &model.ClusterRegResponse{}, err
-	}
-
-	log.Print("NEW CLUSTER REGISTERED : ID-", newCluster.ClusterID, " PID-", newCluster.ProjectID)
```
Comment (gdsoumya) [inline litmus-portal/graphql-server/pkg/graphql/mutations/mutation.go:43]: We should keep this for debugging purpose

**Review by gdsoumya — CHANGES_REQUESTED** [review]

File: litmus-portal/graphql-server/pkg/graphql/mutations/mutation.go (hunk 1 above)
Comment (rajdas98) [reply litmus-portal/graphql-server/pkg/graphql/mutations/mutation.go:43]: It is there just change the flow

Comment (oumkale) [reply litmus-portal/graphql-server/pkg/graphql/mutations/mutation.go:43]: @gdsoumya We need it for showing the details of the cluster on target screens.

**Comment by rajdas98** [comment]
> Why do we need to save the token?

Because the user needs to see the YAML of the not-confirmed cluster. If we don't store it, we will lose it after login and logout

**Review by oumkale — APPROVED** [review]

**Comment by gdsoumya** [comment]
> > Why do we need to save the token?
> 
> Because the user needs to see the YAML of the not-confirmed cluster. If we don't store it, we will lose it after logout

But if the cluster has been confirmed the token is useless, we should delete or make it empty string after cluster confirm

**Comment by oumkale** [comment]
In this release, we don't have a re-connect option. So if it is needed then we have a "details" option where the token will be shown so that he can use token. cc, @rajdas98 @gdsoumya

**Review by gdsoumya — DISMISSED** [review]
LGTM

**Review by gdsoumya — APPROVED** [review]

**Review by oumkale — APPROVED** [review]

**Review by ishangupta-ds — APPROVED** [review]

**Pull request merged**